js/script.js

```diff
@@ -218,3 +220,52 @@ const form = document.getElementById('myform');
 form.addEventListener('submit', validate);
 
 //* **end of form validation code***
+
+//* **preserve data in browser local storage ***
+
+/**
+ * Declaring form elements i-e name, email, textarea
+ * The form elements (fullNameInput, emailInput, and textInput)
+ * are declared by selecting the corresponding HTML elements using document.querySelector().
+*/
+const fullNameInput = document.querySelector('#name');
+const emailInput = document.querySelector('#email');
+const textInput = document.querySelector('#textarea');
+
+// Load the data from local storage, if it already exists in the local storage
+const savedData = JSON.parse(localStorage.getItem('myFormData'));
+
+/**
+ * An if statement checks if there is any saved data (savedData). If data exists,
+ * it populates the form fields (fullNameInput, emailInput, and textInput) with the saved values.
+*/
+
+if (savedData) {
+  fullNameInput.value = savedData.name;
+  emailInput.value = savedData.email;
+  textInput.value = savedData.message;
+}
+/**
+ * The saveFormData() function is defined to capture changes in the
+ *  form input fields. Inside the function, a JavaScript object (myFormData)
+ *  is created with the current values of the input fields.
+ */
+function saveFormData() {
+  const myFormData = {
+    name: fullNameInput.value,
+    email: emailInput.value,
+    text: textInput.value,
+  };
+
+  /**
+  * The myFormData object is then saved to local storage as a single entry
+  *  using localStorage.setItem('myFormData', JSON.stringify(myFormData)).
+  * The object is converted to a JSON string using JSON.stringify() before storing it.
+ */
+
+  localStorage.setItem('myFormData', JSON.stringify(myFormData));
+}
+// adding eventListeners to the form input elements
+fullNameInput.addEventListener('input', saveFormData);
+emailInput.addEventListener('input', saveFormData);
+textInput.addEventListener('input', saveFormData);
```

Comment:
- [ ] Good :+1: job as your application is working as expected. However, I noticed that you have a broken image at the top of your README file. I recommend you fix it maybe bu checking its path or remove it completely from README.
**NOTICE:** In the case you decide to use a logo, make sure you use your own logo and not the Microverse one, if not just remove the image code and only leave the project title.
![Screenshot from 2023-05-19 19-56-13](https://github.com/Mike111222/MyPortfolio/assets/83042742/217f00cc-b047-44b8-ba20-d28dc2fd9caf)